nuxeo-services/nuxeo-permissions/src/main/java/org/nuxeo/ecm/permissions/TransientUserPermissionHelper.java

```diff
@@ -19,34 +19,95 @@
 
 package org.nuxeo.ecm.permissions;
 
+import org.nuxeo.ecm.core.api.CoreInstance;
 import org.nuxeo.ecm.core.api.DocumentModel;
 import org.nuxeo.ecm.core.api.NuxeoPrincipal;
 import org.nuxeo.ecm.core.api.security.ACE;
 import org.nuxeo.ecm.core.api.security.ACL;
 import org.nuxeo.ecm.core.api.security.ACP;
+import org.nuxeo.ecm.core.query.sql.NXQL;
 import org.nuxeo.ecm.tokenauth.service.TokenAuthenticationService;
 import org.nuxeo.runtime.api.Framework;
+import org.nuxeo.runtime.services.config.ConfigurationService;
 
 /**
  * @since 8.1
  */
 public class TransientUserPermissionHelper {
 
+    /**
+     * @since 10.3
+     */
+    public static final String OTHER_DOCUMENT_WITH_PERMISSION_QUERY = "SELECT ecm:uuid FROM Document"
```

Comment:
If you really want to query everything from the database you could do `FROM Document, Relation`. In case there are relations with relevant ACLs.